xarray/plot/plot.py

```diff
@@ -941,3 +1013,82 @@ def pcolormesh(x, y, z, ax, infer_intervals=None, **kwargs):
         ax.set_ylim(y[0], y[-1])
 
     return primitive
+
+
+def scatter(ds, x, y, hue=None, col=None, row=None,
+            col_wrap=None, sharex=True, sharey=True, aspect=None,
+            size=None, subplot_kws=None, add_legend=None,
+            discrete_legend=None, **kwargs):
```

Comment:
I don't know if `discrete_legend` is the best name. I think I prefer `add_colorbar`